packages/framework-core/src/services/read-model-store.ts

```diff
@@ -132,6 +135,61 @@ export class ReadModelStore {
     }
   }
 
+  private getProjections(
+    deletedProjection: boolean,
+    entitySnapshotEnvelope: EntitySnapshotEnvelope
+  ): Array<ProjectionMetadata<EntityInterface, ReadModelInterface>> {
+    if (deletedProjection) {
+      const unProjections: Array<ProjectionMetadata<EntityInterface, ReadModelInterface>> =
+        this.entityUnProjections(entitySnapshotEnvelope)
+      const projections: Array<ProjectionMetadata<EntityInterface, ReadModelInterface>> =
+        this.entityProjections(entitySnapshotEnvelope)
+      if (projections?.length > 0) {
+        if (!unProjections) {
+          throw new Error(`Missing UnProjections for entity ${entitySnapshotEnvelope.entityTypeName}`)
+        }
+        const missingProjection = this.findFirstMissingProjection(projections, unProjections)
+        if (missingProjection) {
+          throw new Error(
+            `Missing UnProjection for ReadModel ${missingProjection.class.name} with joinKey ${missingProjection.joinKey} for entity ${entitySnapshotEnvelope.entityTypeName}`
+          )
+        }
+      }
+      return unProjections
+    }
+    return this.entityProjections(entitySnapshotEnvelope)
+  }
```

Comment:
this function has 2 responsibilities and they depend on a boolean argument. I would split this function into two, one called `getUnprojections(entitySnapshotEnvelope: EntitySnapshotEnvelope)`  and the other one would be the one you have `this.entityProjections(entitySnapshotEnvelope: EntitySnapshotEnvelope)`. From the caller function I'll use this clause to call one or the other: `const projections = deleteEvent ? this.getUnProjections(entitySnapshotEnvelope) : this.entityProjections(entitySnapshotEnvelope)`

Reply:
Good suggestion. Applied in commit 200c434.